product_attribute_variant_rules/views/product_template_views.xml

```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+  <record id="product_template_only_form_view" model="ir.ui.view">
+    <field name="model">product.template</field>
+    <field name="inherit_id" ref="product.product_template_only_form_view" />
+    <field name="arch" type="xml">
+
+      <xpath expr="//page[@name='variants']" position="inside">
+
+        <field name="use_attribute_rules" attrs="{'readonly': [('id','=', False)]}" />
+        <label for="use_attribute_rules" string="Use Attribute Rules" class="mt-3" />
+
+        <!-- FIXME: Sadly the many2many are empty at product creation -->
+        <p class="oe_grey oe_edit_only" attrs="{'readonly': [('id','=', True)]}">
+            Please save the product to be able to edit the attribute rules.
+        </p>
+        <field
+                    name="product_attribute_rule_ids"
+                    attrs="{'invisible': ['|', ('use_attribute_rules', '=', False), ('id','=', False)]}"
+                >
+
+          <tree string="Product Attribute Rules" editable="bottom">
+            <field name="available_precondition_attribute_ids" invisible="1" />
+            <field name="available_postcondition_attribute_ids" invisible="1" />
+            <field
+                            name="product_attribute_value_precondition_ids"
+                            widget="many2many_tags"
+                            domain="[('attribute_id', 'in', available_precondition_attribute_ids)]"
```

Comment:
Domain is not complete here.
we can not filter only on attribute_id.

For example you can have a attribute color with blue, red, green, yellow.

But on the product you only have added yellow and red.
You should not be able to select blue and green

Reply:
Indeed